From: Guenter Roeck <linux@roeck-us.net>
Date: Tue, 2 Apr 2013 09:35:07 +0000
Subject: [PATCH] of_net.h: Provide empty functions if OF_NET is not configured
Patch-mainline: v3.10-rc1
Git-commit: 65b3841b9cb5fe1b239f12dbf033f9827d73d032
References: bsc#909566 FATE#314306

of_get_mac_address() and of_get_phy_mode() are only provided if OF_NET
is configured. While most callers check for the define, not all do, and those
who do require #ifdef around the code. For those who don't, the missing check
can result in errors such as

arch/powerpc/sysdev/tsi108_dev.c:107:3: error: implicit declaration of
	function 'of_get_mac_address' [-Werror=implicit-function-declaration]
arch/powerpc/sysdev/mv64x60_dev.c:253:2: error: implicit declaration of
	function 'of_get_mac_address' [-Werror=implicit-function-declaration]

Provide empty functions if OF_NET is not configured.
This is safe because all callers do check the return values.

Cc: David Daney <david.daney@cavium.com>
Signed-off-by: Guenter Roeck <linux@roeck-us.net>
Acked-by: Rob Herring <rob.herring@calxeda.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Gary Ching-Pang Lin <glin@suse.com>
---
 include/linux/of_net.h |   10 ++++++++++
 1 file changed, 10 insertions(+)

--- a/include/linux/of_net.h
+++ b/include/linux/of_net.h
@@ -10,6 +10,16 @@
 #ifdef CONFIG_OF_NET
 #include <linux/of.h>
 extern const void *of_get_mac_address(struct device_node *np);
+#else
+static inline const int of_get_phy_mode(struct device_node *np)
+{
+	return -ENODEV;
+}
+
+static inline const void *of_get_mac_address(struct device_node *np)
+{
+	return NULL;
+}
 #endif
 
 #endif /* __LINUX_OF_NET_H */
